egs2/jsut/asr1/local/data.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+set -e
+set -u
+set -o pipefail
+
+log() {
+    local fname=${BASH_SOURCE[1]##*/}
+    echo -e "$(date '+%Y-%m-%dT%H:%M:%S') (${fname}:${BASH_LINENO[0]}:${FUNCNAME[1]}) $*"
+}
+SECONDS=0
+
+stage=-1
+stop_stage=1
+sampling_rate=48000
```

Comment:
Please let me to explain of my thought.

- If using format_wav_scp.py with wav.scp including pipe way, e.g. `sox a.wav -r 16k |`, the wav files are recreated even if the sampling rate is not changed. 
   - This is not problem if feats_type=fbank_pitch.
- format_wav.scp itself has `--fs` option and it can changes the sampling rate. Current recipe changes the rate to 16k.
(I don't know the original fs of JSUT, is it 48k?)

Would you change as following?

- If  the sampling rate is not changed from the original rate, then sox isn't used.
- Unify --sampling_rate and --pitch in local/*.sh to --fs
- Add the following arguments to asr.sh

```bash
fs=48000
if [ "${fs}" -eq 48000 ]; then
    # To suppress recreation, specify wav format
    opts=--audio_format wav --fs ${fs} --local_data_opts "--fs ${fs}"
else
    opts=--audio_format flac --fs ${fs} --local_data_opts "--fs ${fs}"
fi
```

Reply:
Sorry, I'm not quite sure what changes I should make for the following points.
> If the sampling rate is not changed from the original rate, then sox isn't used.
Do I need to update `format_wav_scp.py` to implement this process?
> Add the following arguments to asr.sh
Are those options for `format_wav_scp.sh` for feats_type=raw in stage 2? Does this mean flac format does not support 48k?

Reply:
> Do I need to update `format_wav_scp.py` to implement this process?

No you don't.

> Are those options for `format_wav_scp.sh` for feats_type=raw in stage 2? Does this mean flac format does not support 48k?

flac supports 48k.

Sorry for my bad comment. 
My intention is to avoid recreation of wav files. format_wav_scp.sh doesn't generate new audio files if satisfying the following items:

- The specified file format is same as the input file.
- The specified sampling frequency is same as the input file.
- `segments` is not existing.
- Pipe is not written in wav.scp.
  - e.g. `sox a.wav - |` is exactlty same as a.wav but it regenerates files. I'd like to avoid this case.

I assumed that JSUT is 48k data and using wav format and not using segments, then we can achieve this with `--fs 48k --audio_format wav`. 
So only you need to do is to make `local/data_prep.sh` not to use pipe in wav.scp and just write the file path in wav.scp if `--fs 48k`.

On the other hand, If fs != 48k, please use sox in wav.scp. For feats_type=fbank_pitch, we still need to embed sox and also specify `--audio_format flac` because it is inevitable to recreate audio files and we should select `flac` to reduce disk usage for feats_type=raw.